composer/modules/server/services/ballerina-launcher/src/main/java/org/ballerinalang/composer/service/ballerina/launcher/service/LaunchManager.java

```diff
@@ -146,91 +145,116 @@ public void run() {
+                            this.updatePort(getServerStartedURL());
+                            line = LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_CLOUD + " "
+                                    + getServerStartedURL();
+                        }
+
+                        // This is to handle local service run use case.
+                        if (line.startsWith(LauncherConstants.SERVER_CONNECTOR_STARTED_AT_HTTP_LOCAL)
+                                && getServerStartedURL() == null) {
+                            this.updatePort(line);
+                            pushMessageToClient(launchSession, LauncherConstants.OUTPUT, LauncherConstants.DATA, line);
+                        } else {
+                            pushMessageToClient(launchSession, LauncherConstants.OUTPUT, LauncherConstants.DATA, line);
+                        }
+                    }
                 }
-
             }
             pushMessageToClient(launchSession, LauncherConstants.EXECUTION_STOPPED, LauncherConstants.INFO,
-                    LauncherConstants.END_MESSAGE);
+                                LauncherConstants.END_MESSAGE);
             LogParser.getLogParserInstance().stopListner();
         } catch (IOException e) {
             logger.error("Error while sending output stream to client.", e);
-        } finally {
-            if (reader != null) {
-                IOUtils.closeQuietly(reader);
-            }
         }
     }
 
     public void streamError() {
-        BufferedReader reader = null;
-        try {
-            reader = new BufferedReader(new InputStreamReader(
-                    this.command.getProgram().getErrorStream(), Charset.defaultCharset()));
-            String line = "";
-            while ((line = reader.readLine()) != null) {
-                if (this.command.isErrorOutputEnabled()) {
-                    pushMessageToClient(launchSession, LauncherConstants.OUTPUT, LauncherConstants.ERROR, line);
+        try (InputStream inputStream = this.command.getProgram().getErrorStream()) {
+            while (this.command != null && this.command.getProgram().isAlive()) {
+                String output = consumeStreamOutput(inputStream);
+                if (!output.isEmpty()) {
+                    String[] lines = output.split(System.lineSeparator());
+                    for (String line : lines) {
+                        if (this.command.isErrorOutputEnabled()) {
+                            pushMessageToClient(launchSession, LauncherConstants.OUTPUT, LauncherConstants.ERROR, line);
+                        }
+                    }
                 }
             }
         } catch (IOException e) {
             logger.error("Error while sending error stream to client.", e);
-        } finally {
-            if (reader != null) {
-                IOUtils.closeQuietly(reader);
+        }
+    }
+
+    public void pushMessageToProcess(String[] input) {
```

Comment:
Let's add a doc comment for the public method

Reply:
Let's make it `private` method